include/btllib/mi_bloom_filter.hpp

```diff
@@ -1,101 +1,73 @@
+
+private:
+  /** Parse a Bloom filter file header. Useful for implementing Bloom filter
+   * variants. */
+  std::shared_ptr<cpptoml::table> parse_header(const std::string& signature);
+};
+/// @endcond
+
 template<typename T>
 class MIBloomFilter
 {
 public:
-  static const T MASK = 1 << (sizeof(T) * 8 - 1);
+  static const T MASK = T(1) << (sizeof(T) * 8 - 1);
   static const T ANTI_MASK = (T)~MASK;
 
-  static const T STRAND = 1 << (sizeof(T) * 8 - 2);
+  static const T STRAND = T(1) << (sizeof(T) * 8 - 2);
   static const T ANTI_STRAND = (T)~STRAND;
 
   static const T ID_MASK = ANTI_STRAND & ANTI_MASK;
 
   static const unsigned BLOCKSIZE = 512;
 
-  // Calculates the per frame probability of a random match for single value
-  static inline double calc_prob_single_frame(double occupancy,
-                                              unsigned hash_num,
-                                              double freq,
-                                              unsigned allowed_misses)
-  {
-    double prob_total = 0.0;
-    for (unsigned i = hash_num - allowed_misses; i <= hash_num; i++) {
-      double prob = n_choose_k(hash_num, i);
-      prob *= pow(occupancy, i);
-      prob *= pow(1.0 - occupancy, hash_num - i);
-      prob *= (1.0 - pow(1.0 - freq, i));
-      prob_total += prob;
-    }
-    return prob_total;
-  }
-
-  static inline double calc_prob_single(double occupancy, double freq)
-  {
-    return occupancy * freq;
-  }
-
-  /*
-   * Returns an a filter size large enough to maintain an occupancy specified
-   */
-  static size_t calc_optimal_size(size_t entries,
-                                  unsigned hash_num,
-                                  double occupancy)
-  {
-    auto non_64_approx_val =
-      size_t(-double(entries) * double(hash_num) / log(1.0 - occupancy));
-    const int magic = 64;
-    return non_64_approx_val + (magic - non_64_approx_val % magic);
-  }
-
-  /*
-   * Inserts a set of hash values into an sdsl bitvector and returns the number
-   * of collisions Thread safe on the bv, though return values will not be the
-   * same run to run
-   */
-  static unsigned insert(sdsl::bit_vector& bv,
-                         const uint64_t* hash_values,
-                         unsigned hash_num)
-  {
-    unsigned colli_count = 0;
-    for (unsigned i = 0; i < hash_num; ++i) {
-      const int magic = 0x3f;
-      const uint64_t pos = hash_values[i] % bv.size();
-      uint64_t* data_index = bv.data() + (pos >> 6); // NOLINT
-      const uint64_t bit_mask_value = (uint64_t)1 << (pos & magic);
-      colli_count +=
-        __sync_fetch_and_or(data_index, bit_mask_value) >> (pos & magic) & 1;
-    }
-    return colli_count;
-  }
-
-  // TODO: include allowed miss in header
   /// @cond HIDDEN_SYMBOLS
-#pragma pack(1) // to maintain consistent values across platforms
+  //#pragma pack(1) // to maintain consistent values across platforms
```

Comment:
You are right. I just didnt want to delete it before having a conversation. I think Justin put it for having consistent memory operations among platforms however clang-tidy didnt like it and I deleted. Just to give heads up, I think this may result in differences among platforms.

Reply:
Ok, we'll keep this in mind if we see such problems. Thanks for mentioning it. I think it's ok for now if our CI passes.